examples/hotrod/pkg/log/spanlogger.go

```diff
@@ -57,99 +58,164 @@ func (sl spanLogger) With(fields ...zapcore.Field) Logger {
 	return spanLogger{logger: sl.logger.With(fields...), span: sl.span, spanFields: sl.spanFields}
 }
 
-func (sl spanLogger) logToSpan(level string, msg string, fields ...zapcore.Field) {
-	// TODO rather than always converting the fields, we could wrap them into a lazy logger
-	fa := fieldAdapter(make([]log.Field, 0, 2+len(fields)))
-	fa = append(fa, log.String("event", msg))
-	fa = append(fa, log.String("level", level))
+func (sl spanLogger) logToSpan(msg string, fields ...zapcore.Field) {
+	sl.span.AddEvent(
+		msg,
+		trace.WithAttributes(otLogFieldsToOTelAttrs(fields)...),
+	)
+}
+
+func otLogFieldsToOTelAttrs(fields []zapcore.Field) []attribute.KeyValue {
```

Comment:
```suggestion
func logFieldsToOTelAttrs(fields []zapcore.Field) []attribute.KeyValue {
```

Reply:
Renamed in: [b1d21a1](https://github.com/jaegertracing/jaeger/pull/4556/commits/b1d21a13dee7d7083ce6873a0f0c1bd9d4557bba)